openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -123,18 +140,44 @@ impl Drop for SharedPoolHandle {
 // TODO SNP: Implement save restore. This means additionally having some sort of
 // restore_alloc method that maps to an existing allocation.
 #[derive(Inspect)]
-pub struct SharedPool {
+pub struct PagePool {
     #[inspect(flatten)]
-    inner: Arc<Mutex<SharedPoolInner>>,
+    inner: Arc<Mutex<PagePoolInner>>,
+    typ: PoolType,
+    next_device_id: AtomicU64,
 }
 
-impl SharedPool {
-    /// Create a shared pool allocator, with the specified memory. The supplied
-    /// memory is assumed to shared (and therefore unaccepted on SNP).
+impl PagePool {
+    /// Create a new private pool allocator, with the specified memory. The
+    /// memory must not be used by any other entity.
+    pub fn new_private_pool(private_pool: &[MemoryRangeWithNode]) -> anyhow::Result<Self> {
+        let mut pages = Vec::new();
+        for range in private_pool {
```

Comment:
Seems like this is copy/paste code that can be refactored.

(Also push vs map+collect?)